### ISSUE TITLE
Use templateProcessor option to patch inline-ng2-template bug

### ISSUE THREAD

**Issue by RoxKilly** [mgechev/angular-seed#1479]
**Steps to reproduce and a minimal demo of the problem**

**1** Use an external component template that has a dollar sign in front of interpolation:
```
${{amount}}
```

**2** Try to run any task that includes the [`inline-ng2-template`](https://github.com/ludohenin/gulp-inline-ng2-template) plugin, such as `build.prod`.  

**Current behavior**
The task will fail because this plugin fails to escape the `$` when it inlines the template. So the result is `${{amount}}` -- typescript will try and fail to parse -- instead of `\${{amount}}`.

**Expected/desired behavior**

[In `tools/tasks/seed/build.js.prod.ts`](https://github.com/mgechev/angular-seed/blob/master/tools/tasks/seed/build.js.prod.ts#L10) We can add a `templateProcessor` argument to the configurations for this plugin. The custom processor would replace `${` with `\${`.  The same should be done wherever this plugin is used. 

I've tried to do it:
```
const INLINE_OPTIONS = {
  ...
  templateProcessor : 
    ( path:string, ext:string, fileContents:string, callback:(err:any,result?:string)=>void )=> 
    {
      try{
        if(ext==='html') fileContents = fileContents.replace( /\$\{/g, '\\${' );
        callback( null, fileContents );
      } catch(e){ callback(e)}
    }
};
```

 but I get a `lodash` error. Someone more experienced than I may have more luck.

**Other information**

[See this issue](https://github.com/ludohenin/gulp-inline-ng2-template/issues/67)

**Comment by mgechev** [comment]
Hey @RoxKilly thanks for reporting the issue. It seems a problem related to [gulp-inline-ng2-template](https://github.com/ludohenin/gulp-inline-ng2-template) not to the seed. Would you take a look at the issue tracker of the project and open this issue if it's not already there.

**Comment by RoxKilly** [comment]
@mgechev  I linked to an open issue on that project at the bottom of the OP. I figured I'd let you know about this because it seemed like an easy patch that the community would benefit from (just changing one config option) but I understand if you don't want to go down that road.

**Comment by mgechev** [comment]
@RoxKilly yes, thanks for sharing your experience! The fix makes sense, would you open a PR with it?

**Comment by RoxKilly** [comment]
@mgechev I'd love to help. I'm really a newbie at the stuff (not just TS but build tools and git) so I figured I would share what I found here for someone more experience to carry that forward. Specifically, although the fix above seems to work, it triggers another error down the line:
```
/node_modules/gulp-template/node_modules/lodash/lodash.js:479
      case 0: return func.call(thisArg);
                          ^
SyntaxError: Unexpected token .
    at /node_modules/gulp-template/node_modules/lodash/lodash.js:14740:16
    at apply (/node_modules/gulp-template/node_modules/lodash/lodash.js:479:27)
    at /node_modules/gulp-template/node_modules/lodash/lodash.js:15124:16
    at apply (/node_modules/gulp-template/node_modules/lodash/lodash.js:481:27)
    at /node_modules/gulp-template/node_modules/lodash/lodash.js:6499:16
    at template (/node_modules/gulp-template/node_modules/lodash/lodash.js:14739:20)
    at DestroyableTransform._transform (/node_modules/gulp-template/index.js:20:14)
    at DestroyableTransform.Transform._read (/node_modules/gulp-template/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:159:10)
    at DestroyableTransform.Readable.read (/node_modules/gulp-template/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:357:10)
    at flow (/node_modules/gulp-template/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:727:26)
```

This  is a sidenote but: what I'd be happy to do is write-up how I was able to integrate `Angular Material 2`; when I picked up this project yesterday, that was one of my biggest concerns, and [this page is empty](https://github.com/mgechev/angular-seed/wiki/Integration-with-AngularMaterial2)

**Comment by gregjacobs** [comment]
Just FYI, I submitted a pull request to fix that issue in [gulp-inline-ng2-template](https://github.com/ludohenin/gulp-inline-ng2-template). Hopefully will be merged soon: https://github.com/ludohenin/gulp-inline-ng2-template/pull/70

**Comment by RoxKilly** [comment]
This issue has been fixed in [`gulp-inline-ng2-template 3.0.2`](https://github.com/ludohenin/gulp-inline-ng2-template/commit/2dd31bb5e73b5c610e655f10aabe293b237c7cf2)

**Comment by mgechev** [comment]
Sounds good, so it should be already in our master then given that we relay on `^3.0.1`.
